cmake/kokkoskernels_tpls.cmake

```diff
@@ -0,0 +1,414 @@
+FUNCTION(kokkoskernels_append_config_line LINE)
+  GLOBAL_APPEND(KOKKOSKERNELS_TPL_EXPORTS "${LINE}")
+ENDFUNCTION()
+
+MACRO(KOKKOSKERNELS_ADD_TPL_OPTION NAME DEFAULT_VALUE DOCSTRING)
+  KOKKOSKERNELS_ADD_OPTION(ENABLE_TPL_${NAME} ${DEFAULT_VALUE} BOOL ${DOCSTRING})
+  KOKKOSKERNELS_ADD_OPTION(${NAME}_DIR "" PATH "Location of ${NAME} instal root")
+  IF (DEFINED TPL_ENABLE_${NAME})
+    IF (TPL_ENABLE_${NAME} AND NOT KOKKOSKERNELS_ENABLE_TPL_${NAME})
+      MESSAGE(WARNING "Overriding KOKKOSKERNELS_ENABLE_TPL_${NAME}=OFF with TPL_ENABLE_${NAME}=ON")
+      SET(KOKKOSKERNELS_ENABLE_TPL_${NAME} ON)
+    ELSEIF(NOT TPL_ENABLE_${NAME} AND KOKKOSKERNELS_ENABLE_TPL_${NAME})
+      MESSAGE(WARNING "Overriding KOKKOSKERNELS_ENABLE_TPL_${NAME}=ON with TPL_ENABLE_${NAME}=OFF")
+      SET(KOKKOSKERNELS_ENABLE_TPL_${NAME} OFF)
+    ENDIF()
+  ENDIF()
+  IF (KOKKOSKERNELS_ENABLE_TPL_${NAME})
+    LIST(APPEND KOKKOSKERNELS_TPL_LIST ${NAME})
+  ENDIF()
+ENDMACRO()
+
+
+MACRO(kokkoskernels_create_imported_tpl NAME)
+  CMAKE_PARSE_ARGUMENTS(TPL
```

Comment:
@jjwilke I'm having link errors with CUBLAS in a Cuda build.
If I check output after the parse by adding this line:
`MESSAGE(STATUS "AFTER PARSE: TPL_UNPARSED_ARGUMENTS = ${TPL_UNPARSED_ARGUMENTS}")`

 I see the following:
`AFTER PARSE: TPL_UNPARSED_ARGUMENTS = LIBRARIES;/projects/sems/install/rhel7-x86_64/sems/compiler/cuda/9.2/base/lib64/libcublas.so`

It looks like there is inconsistency in the FindTPLCUBLAS.cmake file when passing args to this marco, `LIBRARIES` is passed in instead of `LIBRARY`. If I change "LIBRARY" in the macro to "LIBRARIES" it parses but causes errors.

Reply:
Making the change in FindTPLCUBLAS to pass `LIBRARY` using `CUDA_CUBLAS_LIBRARY` as library path parses correctly and configures, will test out a rebuild.

Reply:
@jjwilke Same linking error with the change I mention above but the macro parse correctly. Should I push the change in to FindTPLCUBLAS (change "LIBRARIES" to "LIBRARY")?

Reply:
`LIBRARIES` should be `LINK_LIBRARIES`